src/udev/udev-builtin-blkid.c

```diff
@@ -321,8 +321,8 @@ static int builtin_blkid(struct udev_device *dev, int argc, char *argv[], bool t
         if (is_gpt)
                 find_gpt_root(dev, pr, test);
 
-        blkid_free_probe(pr);
 out:
+        blkid_free_probe(pr);
```

Comment:
I am not sure how d47f6ca introduced the memory leak since the free was previously before the out and there were other conditions which caused the function to goto that label.

As far as #6110 I am the one who filed that and would be happy to provide a patch. I just wasn't sure if it was more desired to just remove the use of BLKID_SUBLKS_BADCSUM or do two passes and only use BLKID_SUBLKS_BADCSUM if the first probe did not find anything.